File: src/code/Game.java
package code;

import api.EdgeData;

import java.util.ArrayList;
import java.util.Iterator;

public class Game {
    public GraphAlgorithms algo;
    public ArrayList<Pokemon> pokemons;


    public Game(GraphAlgorithms algo) {
        this.algo = new GraphAlgorithms();
        this.algo.init(algo.getGraph());
        pokemons = new ArrayList<>();
    }

    public void setPokemonEdges() {
        for (Pokemon p : pokemons) {
            Iterator<EdgeData> iter = this.algo.getGraph().edgeIter();
            while(iter.hasNext())
            {
                Edge curr = (Edge) iter.next();
                if (curr.getDest() > curr.getSrc() && p.type > 0) {
                    if (curr.formula.checkOnEdge(p.pos)) {
                        p.on = curr;
                        break;
                    }
                } else if (curr.getDest() < curr.getSrc()  && p.type < 0) {
                    if (curr.formula.checkOnEdge(p.pos)) {
                        p.on = curr;
                        break;
                    }
                }
            }

        }
    }
}

File: code/Game.py
from GraphAlgo import GraphAlgo


class Game:
    def __init__(self, graph: GraphAlgo):
        self.graph = GraphAlgo(graph.get_graph())
        self.pokemons = []

    def setPokemonsEdges(self):
        for p in self.pokemons:
            for edge in self.graph.get_graph().get_edges():
                if edge.getDestNode() > edge.getSrcNode() and p.type > 0:
                    if edge.formula.onEdge(p.pos):
                        p.on = edge
                        break
                elif edge.getDestNode() < edge.getSrcNode() and p.type < 0:
                    if edge.formula.onEdge(p.pos):
                        p.on = edge
                        break